### PR TITLE
fix: disallow inviting Admin user if team-level roles are enabled

### PR DESCRIPTION
**Pull request by oioki**
The "Admin" role is retired for paid plans (i.e. if team-level roles are enabled for the org), so it does make sense to remove this role from the role list on the Invite New Members page:
image

Also do backend check in case of frontend bypass.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_member/details.py
Comment (RyanSkonnord) [inline src/sentry/api/endpoints/organization_member/details.py:187]: Ideally I'd prefer to make only one call to `get_allowed_org_roles` and rely on the `Role.is_retired` field to distinguish between the two cases afterward. Not for efficiency's sake, but to avoid the cognitive overhead of two sets of roles.

I have a few ideas about how to refactor this, but it's not worth holding up the fix over. But I might make a quick pass after this is merged?

**Review by RyanSkonnord — APPROVED** [review]

File: src/sentry/api/endpoints/organization_member/__init__.py
Comment (sentaur-athena) [inline src/sentry/api/endpoints/organization_member/__init__.py:97]: Why is this needed? Is there a case that someone wants the retired roles?

File: src/sentry/api/endpoints/organization_member/details.py
Comment (sentaur-athena) [reply src/sentry/api/endpoints/organization_member/details.py:187]: +1

File: src/sentry/api/endpoints/organization_member/index.py
Comment (oioki) [inline src/sentry/api/endpoints/organization_member/index.py:145]: Currently `allow_retired_roles` defaults to True but I feel like it will be useful to reuse this option in other places where `OrganizationMemberRequestSerializer` is invoked. For example, this one could be moved to the validator but in a separate PR:
https://github.com/getsentry/sentry/blob/b36f3f9a51c32e0c156c7beaaec207bbb08623f7/src/sentry/api/endpoints/organization_member/details.py#L298-L306

**Review by RyanSkonnord — APPROVED** [review]
👍 Much cleaner this way.

**Pull request merged**
